dotexporter.py

```diff
@@ -87,23 +96,116 @@ def do_GET(self):
         chain_getHeader = self.query("chain_getHeader")
         system_health   = self.query("system_health")
         runtime_version = self.query("state_getRuntimeVersion")
+        current_best = int(chain_getHeader['number'], 16)
+        best_drift = int(datetime.utcnow().timestamp()- self.previous['best_epoch'])
+
+        if current_best > self.previous['best']:
+           self.previous['best'] = current_best
+
+           self.previous['best_epoch'] = int(datetime.utcnow().timestamp())
+           if DEBUG:
+             print("datetime.utcnow().timestamp()" + str(datetime.utcnow().timestamp()))
+           m.append({
+             'name': 'dot_chain_stall',
+             'prop': { 'block': 'best'},
+             'value': 0
+           })
+        else:
+           if DEBUG:
+             print("self.previous['best_epoch'" + str(self.previous['best_epoch']))
+             print("datetime.utcnow().timestamp()" + str(datetime.utcnow().timestamp()))
+             print("best seconds difference = " + str(int(datetime.utcnow().timestamp())- self.previous['best_epoch']))
+           if((datetime.utcnow().timestamp() - self.previous['best_epoch']) > TIMEOUT):
+             if DEBUG:
+               print("stall detected for best block")
+             m.append({
+              'name': 'dot_chain_stall',
+              'prop': { 'block': 'best'},
+              'value': 1
+              })
+           else:
+             m.append({
+               'name': 'dot_chain_stall',
+               'prop': { 'block': 'best'},
+               'value': 0
+             })
+
 
         try:
           chain_getFinalizedHead   = self.query("chain_getFinalizedHead")
           chain_FinalizedHeadBlock = self.query("chain_getBlock", [chain_getFinalizedHead])
-          m.append({
-            'name': 'dot_chain_block_number',
-            'prop': { 'block': 'finalized' },
-            'value': int(chain_FinalizedHeadBlock['block']['header']['number'], 16)
-          })
+          self.current_finalized = int(chain_FinalizedHeadBlock['block']['header']['number'], 16)
+          final_drift=int(datetime.utcnow().timestamp()- self.previous['finalized_epoch'])
+
+          if self.current_finalized > self.previous['finalized']:
+            self.previous['finalized'] = self.current_finalized
+
+            if(datetime.utcnow().timestamp() >= self.previous['finalized_epoch']):
+                self.previous['finalized_epoch'] = int(datetime.utcnow().timestamp())
+                if DEBUG:
+                  print("finalized: datetime.utcnow().timestamp()" + str(datetime.utcnow().timestamp()))
+                m.append({
+                  'name': 'dot_chain_block_number',
```

Comment:
why is that block here and not below for the best block? the metric will be missing if the finality stalls.